### PR TITLE
[FIX] website: don't try to convert static url in multilang

### PR DESCRIPTION
**Pull request by JKE-be**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JKE-be** [comment]
@odony it should improve perf, but small change in stable version; So 14.0 ? master ?
IMO, 14.0 should be ok since we always considered static file like file that can be served by nginx or other system.Already some special case like in case of 404 we don't check redirect and co...

With this commit, we don't try to find a controller that match the url as soon as /static/ is found in url.

**Comment by odony** [comment]
> IMO, 14.0 should be ok since we always considered static file like file that can be served by nginx or other system.Already some special case like in case of 404 we don't check redirect and co...
> 
> With this commit, we don't try to find a controller that match the url as soon as /static/ is found in url.

Ok for 14.0 (though you can move the `if static` higher up in the function), as this patch has been battle-tested on odoo.com during OXP 2020, and doesn't carry much risk :)

**Comment by JKE-be** [comment]
@robodoo rebase-ff

**Comment by JKE-be** [comment]
Kindly awaiting for approval to merge it and handle fwdport and merge url_rewrite functions  in 14.1

**Review by rdeodoo — APPROVED** [review]

File: addons/http_routing/models/ir_http.py [hunk 1]
```diff
@@ -217,26 +217,24 @@ def is_multilang_url(local_url, lang_url_codes=None):
 
     url = local_url.partition('#')[0].split('?')
     path = url[0]
+
+    # Consider /static/ and /web/ files as non-multilang
+    if '/static/' in path or path.startswith('/web/'):
+        return False
+
     query_string = url[1] if len(url) > 1 else None
     router = request.httprequest.app.get_db_router(request.db).bind('')
```
Comment (odony) [inline addons/http_routing/models/ir_http.py:226]: this line can now be dropped so we save the double bind(), no?

File: addons/http_routing/models/ir_http.py [hunk 2]
```diff
@@ -217,26 +217,24 @@ def is_multilang_url(local_url, lang_url_codes=None):
...
-        func = router.match(path, method='GET', query_args=query_string)[0]
-        return is_multilang_func(func)
-    except werkzeug.exceptions.NotFound:
-        # Consider /static/ files as non-multilang
-        static_index = path.find('/static/', 1)
-        if static_index != -1 and static_index == path.find('/', 1):
-            return False
-        return True
-    except Exception:
+        new_url, func = request.env['ir.http']._url_rewrite_kargs(path, query_args=query_string)
```
Comment (odony) [inline addons/http_routing/models/ir_http.py:234]: when a return value is unused, it is customary to name it `_` to indicate so (and reduce the noise), unless that would be shadowing the translation function, in which case any short name with an `_` prefix works :-)

```suggestion
        _, func = request.env['ir.http']._url_rewrite_kargs(path, query_args=query_string)
```

File: addons/http_routing/models/ir_http.py [hunk 3]
```diff
@@ -217,26 +217,24 @@ def is_multilang_url(local_url, lang_url_codes=None):
...
+        return is_multilang_func(func) if func else True  # /page/xxx has not endpoint/func but multilang
```
Comment (odony) [inline addons/http_routing/models/ir_http.py:235]: `is_multilang_func()` is now used only once, so you could inline it here perhaps?

```suggestion
        # /page/xxx has no endpoint/func but multilang
        return (not func or (
            func.routing.get('website') and
            func.routing.get('multilang', func.routing['type'] == 'http')))
```

**Review by odony — APPROVED** [review]
Looks fine as discussed :) Just a few minor cosmetic remarks, but feel free to ignore them ;-)

**Review by JKE-be — COMMENTED** [review]
I have just updated following your guidelines.
Thanks for the review.

Could you r+ if it is ok for you now ?

**Comment by JKE-be** [comment]
@robodoo r+

**Pull request closed without merging**